packages/bus-dereference/test/ActorDereferenceParse-test.ts

```diff
@@ -0,0 +1,173 @@
+    });
+  });
+
+  it('Should resolve media mappings correctly (unknown extension - given mediaType)', async() => {
+    context = new ActionContext({ });
+    const output = await actor.run({ url: 'https://www.google.com/', context, mediaType: 'rdf' });
+    expect(output.url).toEqual('https://www.google.com/index.html');
+    expect(actor.mediatorParse.mediate).toHaveBeenCalledWith({
+      context,
+      handle: expect.anything(),
+      handleMediaType: 'rdf',
+    });
+  });
+
+  it('Should resolve media mappings correctly (known extension)', async() => {
+    context = new ActionContext({ extension: 'other.x' });
+    const output = await actor.run({ url: 'https://www.google.com/', context });
+    expect(output.url).toEqual('https://www.google.com/other.x');
+    expect(actor.mediatorParse.mediate).toHaveBeenCalledWith({
+      context,
+      handle: expect.anything(),
+      handleMediaType: 'y',
+    });
+  });
+
+  it('should run and receive parse errors', async() => {
+    context = new ActionContext({ emitParseError: true });
+    const output = await actor.run({ url: 'https://www.google.com/', context });
+    expect(output.url).toEqual('https://www.google.com/index.html');
+    await expect(arrayifyStream(output.data)).rejects.toThrow(new Error('Parse error'));
+  });
+
+  it('should run and ignore parse errors in lenient mode', async() => {
+    context = new ActionContext({ emitParseError: true, [KeysInitQuery.lenient.name]: true });
+    const spy = jest.spyOn(actor, <any> 'logError');
+    const output = await actor.run({ url: 'https://www.google.com/', context });
+    expect(output.url).toEqual('https://www.google.com/index.html');
+    expect(await arrayifyStream(output.data)).toEqual([]);
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should run and ignore parse errors in lenient mode and log them', async() => {
+    const logger = new LoggerVoid();
+    const spy = jest.spyOn(logger, 'error');
+    context = new ActionContext({
+      emitParseError: true,
+      [KeysInitQuery.lenient.name]: true,
+      [KeysCore.log.name]: logger,
+    });
+    const output = await actor.run({ url: 'https://www.google.com/', context });
+    expect(await arrayifyStream(output.data)).toEqual([]);
+    expect(spy).toHaveBeenCalledWith('Parse error', {
+      actor: 'actor',
+      url: 'https://www.google.com/',
+    });
+  });
+
+  it('should not run on parse rejects', () => {
+    context = new ActionContext({ parseReject: true });
+    return expect(actor.run({ url: 'https://www.google.com/', context }))
+      .rejects.toThrow(new Error('Parse reject error'));
+  });
+
+  it('should run and ignore parse rejects in lenient mode', async() => {
+    context = new ActionContext({ parseReject: true, [KeysInitQuery.lenient.name]: true });
+    const spy = jest.spyOn(actor, <any> 'logError');
+    const output = await actor.run({ url: 'https://www.google.com/', context });
+    // TODO: check if this needs to pass
```

Comment:
TODO

Reply:
Ah yes - that was to do with this question https://github.com/comunica/comunica/pull/918#issue-1120353896.

The rest of the TODOS I will address tonight - wanted to make sure there were no more major changes before completing them :)